src/integration/java/com/salesforce/dynamodbv2/mt/admin/AmazonDynamoDbAdminUtilsIt.java

```diff
@@ -0,0 +1,122 @@
+import com.amazonaws.services.dynamodbv2.model.ResourceInUseException;
+import com.amazonaws.services.dynamodbv2.util.TableUtils;
+import com.salesforce.dynamodbv2.dynamodblocal.AmazonDynamoDbLocal;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class AmazonDynamoDbAdminUtilsIt {
+
+    /* This tests against hosted DynamoDB to ensure tables encounter TableInUse exception
+    (local create times are too short) */
+
+    static final Regions REGION = Regions.US_EAST_1;
+    AmazonDynamoDB remoteDynamoDB = AmazonDynamoDBClientBuilder.standard()
+            .withCredentials(new EnvironmentVariableCredentialsProvider())
+            .withRegion(REGION).build();
+
+    AmazonDynamoDbAdminUtils remoteUtils = new AmazonDynamoDbAdminUtils(remoteDynamoDB);
+
+    AmazonDynamoDB localDynamoDB = AmazonDynamoDbLocal.getAmazonDynamoDbLocal();
+    AmazonDynamoDbAdminUtils localUtils = new AmazonDynamoDbAdminUtils(localDynamoDB);
+    private static final String TABLE_PREFIX = "oktodelete-testBillingMode.";
+    String tableName;
+    String fullTableName;
+
+    @BeforeEach
+    void beforeEach() {
+        tableName = new String(String.valueOf(System.currentTimeMillis()));
+        fullTableName = TABLE_PREFIX + tableName;
+    }
+
+    @Test
+    void createTableIfNotExists() {
+    }
+
+    private CreateTableRequest getTestCreateTableRequest(String tableName) {
+
+        return new CreateTableRequest()
+                .withTableName(tableName)
+                .withAttributeDefinitions(new AttributeDefinition(HASH_KEY_FIELD, S),
+                        new AttributeDefinition(RANGE_KEY_FIELD, S),
+                        new AttributeDefinition(INDEX_FIELD, S))
+                .withKeySchema(new KeySchemaElement(HASH_KEY_FIELD, KeyType.HASH),
+                        new KeySchemaElement(RANGE_KEY_FIELD, KeyType.RANGE))
+                .withProvisionedThroughput(new ProvisionedThroughput(1L, 1L))
+                .withGlobalSecondaryIndexes(new GlobalSecondaryIndex().withIndexName("testgsi")
+                        .withKeySchema(new KeySchemaElement(INDEX_FIELD, KeyType.HASH))
+                        .withProvisionedThroughput(new ProvisionedThroughput(1L, 1L))
+                        .withProjection(new Projection().withProjectionType(ProjectionType.ALL)))
+                .withLocalSecondaryIndexes(new LocalSecondaryIndex().withIndexName("testlsi")
+                        .withKeySchema(new KeySchemaElement(HASH_KEY_FIELD, KeyType.HASH),
+                                new KeySchemaElement(INDEX_FIELD, KeyType.RANGE))
+                        .withProjection(new Projection().withProjectionType(ProjectionType.ALL)));
+    }
+
+    @Test
+    void createTableIfNotExistsButTableInUseForCreating() {
+
+        remoteDynamoDB.createTable(getTestCreateTableRequest(fullTableName));
+        remoteUtils.createTableIfNotExists(getTestCreateTableRequest(fullTableName), 10);
```

Comment:
Given the name of the table being used, this test will be subject to race conditions if it is running more than once against a given AWS instance correct?

Reply:
Sorry, I now see that you're suffixing with timestamp.  The problem with this is if the test bails, then you leave behind a table that will never be cleaned up by subsequent test executions.  I suggest either ... 1) having the clean up find an purge all tables that start with table prefix(that way those tables eventually get cleaned up), or 2) including the hostname of the machine executing the test and cleaning up afterwards.

Reply:
good point, I will have a "just in case" cleanup step at the end. If each test runs through, they each have a cleanup line depending on the test.

Reply:
ended up moving all cleanup work to afterAll